src/scan_iterator.h

```diff
@@ -37,16 +37,27 @@ ts_scan_iterator_tuple_info(ScanIterator *iterator)
 	return iterator->tinfo;
 }
 
+static inline TupleTableSlot *
+ts_scan_iterator_slot(ScanIterator *iterator)
+{
+	return iterator->tinfo->slot;
+}
+
 static inline HeapTuple
 ts_scan_iterator_tuple(ScanIterator *iterator)
 {
-	return iterator->tinfo->tuple;
+	return ExecFetchSlotHeapTuple(iterator->tinfo->slot, false, NULL);
```

Comment:
Since `ts_scan_iterator_slot` is defined, shouldn't it be used already here?
```suggestion
	return ExecFetchSlotHeapTuple(ts_scan_iterator_slot(iterator), false, NULL);
```